coalib/collecting/Importers.py

```diff
@@ -28,7 +28,14 @@ def _import_module(file_path):
                 module_name = cased_module_name
                 break
 
-    return __import__(module_name)
+    module = __import__(module_name)
+
+    # Change `module_dir` index in `sys.path` to the last index, so bear's
+    # dependencies don't clash with coala-bears packages that have the same
+    # name with it.
+    sys.path.append(sys.path.pop(sys.path.index(module_dir)))
```

Comment:
the `if` on line 18 needs to be removed, so that `module_dir` is always forced to the front of the list, so that another item in the path cant be confused with the `module_dir`.

Maybe also try : first "assert" that `module_dir` is not in the sys.path, and issue a really strong warning , force insertion at the front of the list, and then remove it here afterwards.

Reply:
By "the front of the list" does it mean after the last index of the list?

Reply:
front of the list = first path that Python uses to find the package name.

Reply:
oh ok, I see